test/unit/api-client-test.js

```diff
@@ -67,51 +252,254 @@ describe("ApiClient", function () {
       assert.deepEqual(["s/1/264f46f9", "a/1/42"], story.cacheKeys(1));
     });
 
-    it("returns cache keys for a collection", function () {
+    it("should return all stories and collection id as cache tags for a manual collection", function () {
       const collection = Collection.build({
         id: "42",
-        "collection-cache-keys": ["sc/1/38586"],
+        "collection-cache-keys": ["c/1/42"],
         items: [
           {
             type: "story",
             story: { id: "264f46f9-e624-4718-b391-45e4bca7aeb6" },
           },
+          {
+            type: "story",
+            story: { id: "1e5acfe4-d8d2-40f8-a8f4-049cf1ac9fb4" },
+          },
         ],
       });
       assert.deepEqual(
-        ["c/1/42", "s/1/264f46f9", "sc/1/38586"],
+        ["c/1/42", "s/1/264f46f9", "s/1/1e5acfe4"],
         collection.cacheKeys(1)
       );
     });
 
-    it("adds nested collections when depth is passed", function () {
+    it("should include all nested stories and collection ids as cache tags for a manual collection", function () {
       const collection = Collection.build({
         id: "42",
-        "collection-cache-keys": ["sc/1/38586"],
+        type: "collection",
+        "collection-cache-keys": ["c/1/42"],
         items: [
           {
+            type: "story",
+            story: { id: "264f46f9-e624-4718-b391-45e4bca7aeb6" },
+          },
+          {
+            type: "story",
+            story: { id: "1e5acfe4-d8d2-40f8-a8f4-049cf1ac9fb4" },
+          },
+          {
+            id: "43",
             type: "collection",
+            "collection-cache-keys": ["c/1/43"],
+            items: [
+              {
+                type: "story",
+                story: { id: "0c2809d8-c60b-4cf2-b5eb-9aaeec90c062" },
+              },
+              {
+                type: "story",
+                story: { id: "fd4fa2c4-5441-434a-bf76-7660bbf1a09d" },
+              },
+            ],
+          }
+        ],
+      });
+      assert.deepEqual(
+        ["c/1/42", "s/1/264f46f9", "s/1/1e5acfe4", "c/1/43", "s/1/0c2809d8", "s/1/fd4fa2c4"],
+        collection.cacheKeys(1)
+      );
+    });
+
+    it("should return only collection cache keys excluding story keys as cache keys for an automated collection", function () {
```

Comment:
So this is the case we're ignoring where the stories are not part of the rules but manually added?

Reply:
Yes. In the case of automated collection, we only show the stories that match the rules condition in LHS. And we can rank stories only if it shows up in the LHS. Though there's a hacky way to bring up stories that doesn't match the filter condition we are not considering that corner case. (Hack: Rank a story and now remove that rule from collection, the story will still be part of the automated collection)